### PR TITLE
Cache devnet on Windows

### PR DESCRIPTION
**Pull request by franciszekjob**
Relates #1399 

## Introduced changes

- Cache devnet binary on Windows to speed up CI
- Enable running Windows tests on any github event (previously running them on pull request was disabled)

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -358,11 +358,19 @@ jobs:
 
       # ====================== SETUP DEVNET ====================== #
 
+      - name: Cache devnet build
```
Comment (ddoktorski) [inline .github/workflows/checks.yml:361]: Let's apply these changes also to `run-docs-tests-windows` below

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -358,11 +358,19 @@ jobs:
...
+        id: windows-devnet-cache
+        uses: actions/cache@v3
```
Comment (ddoktorski) [inline .github/workflows/checks.yml:363]: Can we use `v4`?

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -329,7 +329,6 @@ jobs:
   # ---------------------------------------------------------- #
 
   run-tests-windows:
-    if: ${{ github.event_name != 'pull_request' }}
```
Comment (ddoktorski) [inline .github/workflows/checks.yml:332]: I think it's still better to skip windows tests on pull requests

**Review by ddoktorski — APPROVED** [review]
💪

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -338,6 +338,7 @@ jobs:
       matrix:
         python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12" ]
     env:
+      DEVNET_SHA: 7e7dbb5
```
[the remark this answers is not in the document]
Comment (franciszekjob) [reply .github/workflows/checks.yml]: Correct, thanks for good catch!

**Pull request merged**